packages/pipeline-editor/src/CustomFormControls/StringControl.tsx

```diff
@@ -130,14 +132,24 @@ function StringControl({
   return (
     <Container className={errorMessages.length > 0 ? "error" : undefined}>
       <InputContainer>
-        <input
-          type="text"
-          value={renderedValue}
-          placeholder={placeholder}
-          onChange={handleChange}
-          disabled={format === "file"}
-          onBlur={handleBlur}
-        />
+        {textarea ? (
+          <textarea
+            value={renderedValue}
+            placeholder={placeholder}
+            onChange={handleChange}
+            onBlur={handleBlur}
+          />
+        ) : (
+          <input
+            type="text"
+            value={renderedValue}
+            placeholder={placeholder}
+            onChange={handleChange}
+            disabled={format === "file"}
+            onBlur={handleBlur}
+            readOnly={readonly}
```

Comment:
Done - should I change the name of this parameter as well? Maybe to something like `displayOnly`?

Reply:
Oh, for the `readonly` part I meant create a new react component, like `DisplayControl.tsx`. That component would just be a super simple component that basically returns the `p` tag. Mainly because we don't need any of the update logic that the StringControl provides. This would also make it a bit easier to add markdown support or other fancy options later on